### ISSUE TITLE
TabBar of TabsController is hidden under NavigationBar on iOS10

### ISSUE THREAD

**Issue by takehilo** [CosmicMind/Material#1237]
on iOS10.3.1, TabBar of TabsController is hidden under NavigationBar.

![image](https://user-images.githubusercontent.com/23430968/58219698-cf437c80-7d46-11e9-8e5c-5131d4c80253.png)

It's fine on iOS11 and iOS12.

![image](https://user-images.githubusercontent.com/23430968/58219714-da96a800-7d46-11e9-86ee-24609af37723.png)

Sample app code:

ViewController.swift
```swift
import UIKit
import Material

class Tab1ViewController: UITableViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        tabItem.title = "Tab1"
    }
}

class Tab2ViewController: UITableViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        tabItem.title = "Tab2"
    }
}

class AppStatusBarController: StatusBarController {
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "Material Tab Test"
        displayStyle = .partial
    }
}

class AppTabsController: TabsController {
    override func viewDidLoad() {
        super.viewDidLoad()
        tabBarAlignment = .top
        tabBar.lineAlignment = .bottom
        tabBar.isDividerHidden = true
    }
}
```

AppDelegate.swift
```swift
import UIKit

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {

    var window: UIWindow?

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        let tab1 = Tab1ViewController()
        let tab2 = Tab2ViewController()
        let tabs = AppTabsController(viewControllers: [tab1, tab2])
        let statusBar = AppStatusBarController(rootViewController: tabs)
        let nav = UINavigationController(rootViewController: statusBar)

        window = UIWindow(frame: UIScreen.main.bounds)
        window?.rootViewController = nav
        window?.makeKeyAndVisible()
        return true
    }
}
```

How can I fix this?
Thank you.

**Comment by OrkhanAlikhanov** [comment]
Hey! Change `UINavigationController` to `NavigationController` and it will work.  Because `NavigationController`'s navigation bar is opaque in contrast to `UINavigationController`'s translucent navigation bar. 

This is because how iOS is designed by default. It will extend view controller underneath the translucent bars. If you set `extendedLayoutIncludesOpaqueBars` to `true` Material's opaque navigation bar will also overlap the child view controller's view.

You can change this behavior altogether by setting `edgesForExtendedLayout` to an empty set (`[]`) or by excluding `.top` edge from set. This is the behavior in iOS 10 and earlier.

Here is what apple documentation says about it:
### [`edgesForExtendedLayout`](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621515-edgesforextendedlayout)
> Instead of this property, use the safe area of your view to determine which parts of your interface are occluded by other content. For more information, see the safeAreaLayoutGuide and safeAreaInsets properties of UIView.
> 
> **In iOS 10 and earlier**, use this property to report which edges of your view controller extend underneath navigation bars or other system-provided views. The default value of this property is all, and it is recommended that you do not change that value.
> 
> If you remove an edge value from this property, the system does not lay out your content underneath other bars on that same edge. In addition, the system provides a default background so that translucent bars have an appropriate appearance. The window’s root view controller does not react to this property.

Following would make it work in your case:
```swift
class AppTabsController: TabsController {
    override func prepare() {
        super.prepare()
        edgesForExtendedLayout = []
    }
}

```

So, basically, this has nothing to do with `Material` :)

**Comment by danieldahan** [comment]
Thank you @OrkhanAlikhanov :) . @takehilo if you have any further questions or issues, please reopen or create a new issue. Thank you!

**Comment by takehilo** [comment]
`edgesForExtendedLayout = []` worked!
Thank you @OrkhanAlikhanov
